privaterelay/settings.py

```diff
@@ -708,3 +708,23 @@ def _get_initial_middleware():
 if USE_SILK:
     SILKY_PYTHON_PROFILER=True
     SILKY_PYTHON_PROFILER_BINARY=True
+
+# Settings for manage.py process_emails_from_sqs
+PROCESS_EMAIL_BATCH_SIZE = config(
+    "PROCESS_EMAIL_BATCH_SIZE", 10, cast=Choices(range(1, 11), cast=int)
+)
+PROCESS_EMAIL_DELETE_FAILED_MESSAGES = config(
+    "PROCESS_EMAIL_DELETE_FAILED_MESSAGES", False, cast=bool
+)
+PROCESS_EMAIL_HEALTHCHECK_PATH = config("PROCESS_EMAIL_HEALTHCHECK_PATH", "") or None
```

Comment:
question & suggestion (blocking): `process_emails_from_sqs` treats the healthcheck file as optional. Should it?

Since `check_health` requires a `healthcheck_path`, let's either add a default value for it, or add a local dev value for it to `.env-dist`.

When I tried to run `python manage.py check_health` I got the error:
```
manage.py check_health: error: the following arguments are required: healthcheck_path
```

I added a value to my `PROCESS_EMAIL_HEALTHCHECK_PATH` env var ...
```bash
> echo $PROCESS_EMAIL_HEALTHCHECK_PATH
/tmp/privaterelay_process_emails_from_sqs_healthcheck
```

But I still got:
```
manage.py check_health: error: the following arguments are required: healthcheck_path
``````

Furthermore, even when I pass a healthcheck_path argument to `check_health` ...
```
python manage.py check_health /tmp/privaterelay_process_emails_from_sqs_healthcheck
```

I got: `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

So I dug into `process_emails_from_sqs.py` and saw that it only has a single call to `write_healthcheck` in the `except ClientError as e` block of code? Did we remove the "healthy" write call?

Reply:
Thanks for the in-depth review!

This is the [second time](https://github.com/mozilla/fx-private-relay/pull/1768#discussion_r847855939) you expected a default for the healthcheck file, so I'll add it.

I'm wondering if we should have any command line parameters for these functions, since you and @moz-astults prefer environment variables. That's a bigger change, but if we tackle it now, then we'll discover bugs while our attention is on the new code.

Good catch on the healthcheck call! I accidentally removed it in https://github.com/mozilla/fx-private-relay/commit/f8f6b737fa4e2b9ec12009f04ade83eb28df976d. I'll add a failing test and fix it.

I also think it makes sense to touch the file after processing each email. A successful call to SES in the dev deployment takes 100 ms. The auto-retry functionality could stretch this into seconds, but shouldn't cause the task to be killed.

> I got: `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

It is wrong that there is no file, but I intentionally did _not_ add a lot of error handling to `check_health`. This error from trying to read an empty file as JSON will exit with a non-zero code, and be (correctly) interpreted as a health failure.